Add m2 keyboard. (#19853)
qmk/qmk_firmware: keyboards/monsgeek/m2/readme.md

Reviewer: An image of this keyboard/PCB would be helpful please 🙂

# monsgeek m2

A customizable 98-key keyboard

* Keyboard Maintainer: [jonylee@hfd](https://github.com/jonylee1986)
* Hardware Supported: monsgeek m2
* Hardware Availability: [monsgeek](https://www.monsgeek.com/)

Make example for this keyboard (after setting up your build environment):

    make monsgeek/m2:default

>     make monsgeek/m2:via

Reviewer 2: ```suggestion
```

> # monsgeek m2
> 
> A customizable 95% keyboard
> 
> * Keyboard Maintainer: [jonylee@hfd](https://github.com/jonylee1986)
> * Hardware Supported: monsgeek m2

Reviewer: ```suggestion
# MonsGeek M2

A customizable 95% keyboard

* Keyboard Maintainer: [jonylee@hfd](https://github.com/jonylee1986)
* Hardware Supported: MonsGeek M2 PCB
```

* Hardware Availability: [monsgeek](https://www.monsgeek.com/)

Make example for this keyboard (after setting up your build environment):

    make monsgeek/m2:default

Flashing example for this keyboard:

    make monsgeek/m2:default:flash

> 
> **Reset Key**: Hold down the key located at *K01*, which programmed as *Esc* while plugging in the keyboard.
> 
> See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).

Reviewer: ```suggestion
See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).

## Bootloader

Enter the bootloader in 2 ways:

* **Reset Key**: Hold down the key located at *K01*, which programmed as *Esc* while plugging in the keyboard.
* **Keycode in layout**: Press the key mapped to `QK_BOOT` if it is available
```

# MonsGeek M2

A customizable 95% keyboard

* Keyboard Maintainer: [jonylee@hfd](https://github.com/jonylee1986)
* Hardware Supported: MonsGeek M2 PCB
* Hardware Availability: [monsgeek](https://www.monsgeek.com/)

Make example for this keyboard (after setting up your build environment):

    make monsgeek/m2:default

Flashing example for this keyboard:

    make monsgeek/m2:default:flash

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).

## Bootloader

Enter the bootloader in 2 ways:

> * **Reset Key**: Hold down the key located at *K01*, which programmed as *Esc* while plugging in the keyboard.

Reviewer 3: ```suggestion
* **Reset Key**: Hold down the top key (commonly programmed as *Esc*) while plugging in the keyboard.
```

# MonsGeek M2

> [![M2](https://s21.ax1x.com/2024/03/23/pFhDAqH.md.png)](https://imgse.com/i/pFhDAqH)

Reviewer 4: Why is the image a click through to an unrelated site?
And why is it hosted on an unknown link,'s21.az1x'—what ` - what is the guarantee of permanence on this? *(The PR guidelines is to host on `imgur`.)*

A customizable 95% keyboard

* Keyboard Maintainer: [jonylee@hfd](https://github.com/jonylee1986)
* Hardware Supported: MonsGeek M2 PCB

> * Hardware Availability: [monsgeek](https://www.monsgeek.com/)

Reviewer 5: A more accurate link would be: https://www.monsgeek.com/product/m2/

Also, doesn't appear to be wireless.

Enter the bootloader in 2 ways:

> * **Reset Key**: Hold down the top key (commonly programmed as *Esc*) while plugging in the keyboard.

Reviewer 3: I mis-typed my previous suggestion. :man_facepalming: 

```suggestion
* **Reset Key**: Hold down the top left key (commonly programmed as *Esc*) while plugging in the keyboard.
```
